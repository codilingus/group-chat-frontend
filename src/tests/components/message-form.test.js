import React from 'react';
import { shallow } from 'enzyme';
import MessageForm from '../../modules/main/message-field/message-form';

test('should render MessageForm component correctly', () => {
  const wrapper = shallow(<MessageForm />);
  expect(wrapper).toMatchSnapshot();
});

test('should set value on input message change', () => {
  const value = 'some message';
  const wrapper = shallow(<MessageForm />);
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('message')).toBe(value);
});
