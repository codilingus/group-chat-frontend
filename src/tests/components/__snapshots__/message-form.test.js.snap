// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MessageForm component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        className="message-form-input"
        onChange={[Function]}
        placeholder="Message radnom"
        value=""
      />,
      "className": "message-form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "message-form-input",
        "onChange": [Function],
        "placeholder": "Message radnom",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          className="message-form-input"
          onChange={[Function]}
          placeholder="Message radnom"
          value=""
        />,
        "className": "message-form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "message-form-input",
          "onChange": [Function],
          "placeholder": "Message radnom",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
